x-pack/plugins/triggers_actions_ui/public/application/components/update_api_key_modal_confirmation.tsx

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { EuiConfirmModal } from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+import React, { useEffect, useState } from 'react';
+import { HttpSetup } from '@kbn/core/public';
+import { useKibana } from '../../common/lib/kibana';
+export const UpdateApiKeyModalConfirmation = ({
+  onCancel,
+  idsToUpdate,
+  apiUpdateApiKeyCall,
+  setIsLoadingState,
+  onUpdated,
+}: {
+  onCancel: () => void;
+  idsToUpdate: string[];
+  apiUpdateApiKeyCall: ({ id, http }: { id: string; http: HttpSetup }) => Promise<string>;
+  setIsLoadingState: (isLoading: boolean) => void;
+  onUpdated: () => void;
+}) => {
+  const {
+    http,
+    notifications: { toasts },
+  } = useKibana().services;
+
+  const [updateModalFlyoutVisible, setUpdateModalVisibility] = useState<boolean>(false);
+
+  useEffect(() => {
+    setUpdateModalVisibility(idsToUpdate.length > 0);
+  }, [idsToUpdate]);
+
+  return updateModalFlyoutVisible ? (
+    <EuiConfirmModal
+      buttonColor="primary"
+      data-test-subj="updateApiKeyIdsConfirmation"
+      title={i18n.translate('xpack.triggersActionsUI.updateApiKeyConfirmModal.title', {
+        defaultMessage: 'Update API Key',
+      })}
+      onCancel={() => {
+        setUpdateModalVisibility(false);
+        onCancel();
+      }}
+      onConfirm={async () => {
+        setUpdateModalVisibility(false);
+        setIsLoadingState(true);
+        try {
+          await apiUpdateApiKeyCall({ id: idsToUpdate[0], http });
```

Comment:
**question:** should this update to all ids?

```suggestion
          await Promise.all(
              idsToUpdate.map((id) => apiUpdateApiKeyCall({ id, http }))
          );
```

Reply:
Currently it can update only one rule at a time.
I thought we can change this part when we add the bulk update support.

Reply:
Gotcha, it feels like we should make the confirm modal work with a single id or add this for loop to support multiple calls (future proof). Thoughts?

Reply:
Good idea, applied :)